File: Ch03/03_02/paths.py
def split_ext(path):
    i = path.rfind('.')
    if i == -1:
        return path, ''
    return path[:i], path[i:]


if __name__ == '__main__':
    root, ext = split_ext('app.py')
    print(f'{root=}, {ext=}')

File: Ch03/03_02/paths.go
package main

import (
	"fmt"
	"strings"
)

func SplitExt(path string) (string, string) {
	i := strings.LastIndex(path, ".")
	if i == -1 {
		return path, ""
	}

	return path[:i], path[i:]
}

func main() {
	root, ext := SplitExt("app.go")
	fmt.Printf("root=%#v, path=%#v\n", root, ext)
}
